sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/Security/Ucfg/MethodId_Get.cs

```diff
@@ -0,0 +1,95 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2018 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+extern alias csharp;
+using csharp::SonarAnalyzer.Security.Ucfg;
+using FluentAssertions;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace SonarAnalyzer.UnitTest.Security.Ucfg
+{
+    [TestClass]
+    public class MethodId_Get
+    {
+        [TestMethod]
+        public void MethodId_Get_Specification()
+        {
+            const string code = @"
+using System;
+namespace Namespace
+{
+    public class Class1
+    {
+        public void PrimitiveTypes1(string s) { }
+        public void PrimitiveTypes2(String s) { }
+        public void PrimitiveTypes3(System.String s) { }
+        public void UserType(Class1 s) { }
+        public void SystemType(Uri uri) { }
+        public void GenericMethod1<T1,T2>(T1 x) { }
+
+        public class Class2
+        {
+            public void InnerClass(Class2 s) { }
+        }
+    }
+    public class GenericClass<T1>
+    {
+        public void GenericMethod2(T1 x) { }
+        public void GenericMethod3<T2>(T1 x, T2 y) { }
+    }
+    public class BaseClass<T1>
+    {
+        public virtual void Method1(T1 x) { }
+        public virtual void Method2(T1 x) { }
+    }
+    public class Descendant : BaseClass<string>
+    {
+        public override void Method1(string x) { }
+    }
+}
+public class Class3
+{
+    public void NoNamespace(Class3 s) { }
+}";
```

Comment:
Other test cases:
* nested namespaces
* structs
* array types
* bound generic types e.g. _IList<Class3>_
* extension methods
* nested methods - even if we can't handle these, we want to know they won't fail

I don't think it's necessary to add all of the test cases now, but may add some TODOs to list the cases that aren't covered.